jib-cli/src/main/java/com/google/cloud/tools/jib/cli/buildfile/FilePropertiesSpec.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.cli.buildfile;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.cloud.tools.jib.api.buildplan.FilePermissions;
+import java.time.Instant;
+import java.util.Optional;
+import javax.annotation.Nullable;
+
+/**
+ * A yaml block for specifying file properties (used in conjunction with layers).
+ *
+ * <p>Example use of this yaml snippet.
+ *
+ * <pre>{@code
+ * filePermissions: 644
+ * directoryPermissions: 755
+ * user: foo
+ * group: bar
+ * timestamp: 100
+ * }</pre>
+ */
+public class FilePropertiesSpec {
+  @Nullable private FilePermissions filePermissions;
+  @Nullable private FilePermissions directoryPermissions;
+  @Nullable private String user;
+  @Nullable private String group;
+  @Nullable private Instant timestamp;
```

Comment:
Yeah actually I think so, I was using a different jackson mechanism to load the values, but since I'm using `@JsonCreator` now, final seems possible, let me test it out.